### PR TITLE
Get result metadata for queries that bypass PS framework

### LINKED ISSUE #67: SHOW PLUGINS returns nothing
Using RMariaDB v1.0-2 with both MariaDB 10.1.25 and MySQL 5.7.17, I'm not able to successfully run the `SHOW PLUGINS` statement (I get empty `df`'s).

```r
> root <- DBI::dbConnect(RMariaDB::MariaDB(), username = "root", password = "***")
> DBI::dbGetQuery(root, "SHOW PLUGINS")
data frame with 0 columns and 0 rows
```
With RMySQL v0.10.13, however, this works fine
```r
> DBI::dbGetQuery(root, "SHOW PLUGINS")
                            Name   Status               Type Library License
1                         binlog   ACTIVE     STORAGE ENGINE    <NA>     GPL
2          mysql_native_password   ACTIVE     AUTHENTICATION    <NA>     GPL
3             mysql_old_password   ACTIVE     AUTHENTICATION    <NA>     GPL
...
```
Please let me know whether I can provide further information for helping to solve this issue.

### PR DESCRIPTION
**Pull request by nbenn**
Some queries, such as SHOW bypass the PS framework, causing `mysql_stmt_result_metadata` not to return any metadata at prepare (c.f. [mysqlnd](https://github.com/php/php-src/blob/39327b9/ext/mysqlnd/mysqlnd_ps.c#L442)). A call to `mysql_stmt_result_metadata` at a later time attempts to remedy this issue.

Fixes #67.

### CODE REVIEW

File: R/query.R [hunk 1]
```diff
@@ -53,20 +56,33 @@ setMethod("dbFetch", "MariaDBResult",
...
 
-    rs <- new("MariaDBResult",
-      sql = statement,
-      ptr = result_create(conn@ptr, statement)
-    )
+#' @rdname query
+#' @export
+setMethod("dbSendStatement", signature("MariaDBConnection", "character"),
+  function(conn, statement, params = NULL, ...) {
+    dbSend(conn, statement, params, TRUE)
```
Comment (krlmlr) [inline R/query.R]: `is_statement = TRUE` instead of `TRUE` is a bit easier to understand. Same above for `FALSE`.

File: R/query.R [hunk 2]
```diff
@@ -53,20 +56,33 @@ setMethod("dbFetch", "MariaDBResult",
...
+  }
+)
+
+dbSend <- function(conn, statement, params = NULL, is.stmnt = FALSE) {
```
Comment (krlmlr) [inline R/query.R]: Do we need default values here?

File: src/MariaResult.h [hunk 3]
```diff
@@ -36,7 +36,7 @@ class MariaResult {
   void autocommit();
 
 public:
-  static MariaResult* create_and_send_query(MariaConnectionPtr con, const std::string& sql);
+  static MariaResult* create_and_send_query(MariaConnectionPtr con, const std::string& sql, bool is_stmnt = false);
```
Comment (krlmlr) [inline src/MariaResult.h]: Default value necessary?

File: src/MariaResult.cpp [hunk 4]
```diff
@@ -34,8 +34,8 @@ void MariaResult::autocommit() {
   maria_conn->autocommit();
 }
 
-MariaResult* MariaResult::create_and_send_query(MariaConnectionPtr con, const std::string& sql) {
-  std::auto_ptr<MariaResult> res(new MariaResultPrep(con));
+MariaResult* MariaResult::create_and_send_query(MariaConnectionPtr con, const std::string& sql, bool is_stmnt) {
+  std::auto_ptr<MariaResult> res(new MariaResultPrep(con, is_stmnt));
```
Comment (krlmlr) [inline src/MariaResult.cpp]: I have yet to encounter a query (with `!is_statement`) that doesn't work with the prepared statement protocol, but if this happens we may need to pass `is_statement` to the `MariaResultSimple()` constructor below. We don't need it now, but a comment to that effect might be worthwhile adding.

Comment (nbenn) [reply src/MariaResult.cpp]: I added a comment, although I have to admit, I do not understand what for. But if it's ok for you like this, it's ok for me 😉.

**Comment by nbenn** [comment]
Thanks for the feedback.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**